executor/src/witgen/rows.rs

```diff
@@ -195,9 +201,63 @@ impl<T: FieldElement> Debug for Cell<'_, T> {
     }
 }
 
-/// A row of cells, indexed by polynomial ID.
 pub type Row<'a, T> = WitnessColumnMap<Cell<'a, T>>;
 
+/// Merges two rows, updating the first.
+/// Range constrainst from the second row are ignored.
+/// @returns false if there was a conflict (two different known values)
+#[must_use]
+pub fn merge_row_with<'a, T: FieldElement>(row: &mut Row<'a, T>, other: &Row<'a, T>) -> bool {
+    // First check for conflicts.
+    if row
+        .values()
+        .zip(other.values())
+        .any(|(cell1, cell2)| match (&cell1.value, &cell2.value) {
+            (CellValue::Known(v1), CellValue::Known(v2)) => v1 != v2,
+            _ => false,
+        })
+    {
+        return false;
+    };
+    *row = WitnessColumnMap::from(row.values().zip(other.values()).map(|(cell1, cell2)| {
+        match (&cell1.value, &cell2.value) {
+            (CellValue::Known(_), _) => cell1.clone(),
+            _ => cell2.clone(),
+        }
+    }));
+    true
+}
+
+// TODO will be impl method of row later.
+pub fn value_is_known<T: FieldElement>(row: &Row<'_, T>, poly_id: &PolyID) -> bool {
+    row[poly_id].value.is_known()
+}
+
+// TODO will be impl method of row later.
+pub fn set_cell_unknown<T: FieldElement>(row: &mut Row<'_, T>, poly_id: &PolyID) {
+    row[poly_id].value = CellValue::Unknown;
+}
+
+// TODO will be impl method of row later.
```

Comment:
What's us from doing that now? Also for the functions below.

Reply:
Row is currently only a typedef.